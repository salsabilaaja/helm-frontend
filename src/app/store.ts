import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import answearReducer from '../features/answearSlice'


export const store = configureStore({
  reducer: {
    answear: answearReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
