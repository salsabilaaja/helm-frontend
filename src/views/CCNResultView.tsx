import React, { useLayoutEffect, useState } from "react"
import { useAppSelector } from "../app/hooks"
import CCNActionButton from "../components/button/CCNActionButton"
import { getAnswearId } from "../features/answearSlice"
import Questionnaire from '../static/data/questionnaire.json'

import dummy      from "../assets/img/dummy.jpg"
import adventura  from "../assets/img/adventura.png"
import newNomad   from "../assets/img/newNomad.png"
import yelos      from "../assets/img/yelos.png"
import zelos      from "../assets/img/zelos.png"
import newzyra    from "../assets/img/newzyra.png"
import newway     from "../assets/img/newway.png"


function CCNResultView() {
  const [result, setResult]                   = useState({ name: "-", voucher: "-" })
  const arrHelmImage                          = [dummy, adventura, newNomad, yelos, zelos, newzyra, newway]
  const [helmImageResult, setHelmImageResult] = useState(dummy)
  const answearId                             = useAppSelector(getAnswearId)


  useLayoutEffect(() => {
    const resultIndex = Questionnaire.listOfResult.findIndex(item => item.id === answearId)
    if(resultIndex !== -1) {
      const result = {
        ...Questionnaire.listOfResult[resultIndex]
      }
      setResult(result)
      getImage(Questionnaire.listOfResult[resultIndex].name)
    }
  }, [])

  const getImage = (value: string) => {
    const resultHelm = value.toLocaleLowerCase()

    for (let index = 0; index < arrHelmImage.length; index++) {
      const helm = arrHelmImage[index].toLowerCase()
      if(helm.includes(resultHelm))
        setHelmImageResult(arrHelmImage[index])
    }
  }


  return (
    <div className="resultContainer">
      <div className="resultContentContainer">
        <div className="resultContent">
          <p className="txtResultTitle">Your helmet are...</p>
          <p className="txtResultHelm">{result.name}</p>
          <p className="txtResultDesc mt-[25px]">Claim your <b>50% discount</b> with this promo code</p>
          <CCNActionButton
            className="mt-[20px] mr-[25px] !w-[175px]"
            a_strText={result.voucher}
            filled
            copyText
          />
        </div>
        <div className="resultPicture">
          <img src={helmImageResult} alt="helmet" />
        </div>
      </div>
    </div>
  )
}

export default CCNResultView
