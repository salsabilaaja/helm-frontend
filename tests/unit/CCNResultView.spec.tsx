import React from 'react'
import { render } from "@testing-library/react"
import { configureStore } from '@reduxjs/toolkit'
import { store } from '../../src/app/store'
import { Provider } from 'react-redux'

import CCNResultView from "../../src/views/CCNResultView"
import * as hooks from '../../src/app/hooks'


describe("CCNResultView.tsx", () => {
  it("Render Properly", async () => {
    const wrapper = render(
      <Provider store={store}>
        <CCNResultView />
      </Provider>
    )
    await expect(wrapper).toMatchSnapshot()
  })

  it('Result should show Newway Helm', () => {
    const useDispatchSpy = jest.spyOn(hooks, 'useAppDispatch')
    const mockDispatchFn = jest.fn()
    useDispatchSpy.mockReturnValue(mockDispatchFn)

    const { container } = render(
      <Provider store={store}>
        <CCNResultView />
      </Provider>
    )

    const txtHelmName: HTMLElement  = container.querySelector('.txtResultHelm') as HTMLElement
    expect(txtHelmName.innerHTML).toBe("Newway")
  })
})
