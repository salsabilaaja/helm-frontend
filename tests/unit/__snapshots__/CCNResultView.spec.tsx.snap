// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCNResultView.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="resultContainer"
      >
        <div
          class="resultContentContainer"
        >
          <div
            class="resultContent"
          >
            <p
              class="txtResultTitle"
            >
              Your helmet are...
            </p>
            <p
              class="txtResultHelm"
            >
              Newway
            </p>
            <p
              class="txtResultDesc mt-[25px]"
            >
              Claim your 
              <b>
                50% discount
              </b>
               with this promo code
            </p>
            <button
              aria-label=""
              class="btnActionContainer px-[10px]
        btnActionFilled 
        mt-[20px] mr-[25px] !w-[175px]"
              data-mui-internal-clone-element="true"
            >
              <p
                class="btnActionText"
              >
                NEWWAY20
              </p>
            </button>
          </div>
          <div
            class="resultPicture"
          >
            <img
              alt="helmet"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="resultContainer"
    >
      <div
        class="resultContentContainer"
      >
        <div
          class="resultContent"
        >
          <p
            class="txtResultTitle"
          >
            Your helmet are...
          </p>
          <p
            class="txtResultHelm"
          >
            Newway
          </p>
          <p
            class="txtResultDesc mt-[25px]"
          >
            Claim your 
            <b>
              50% discount
            </b>
             with this promo code
          </p>
          <button
            aria-label=""
            class="btnActionContainer px-[10px]
        btnActionFilled 
        mt-[20px] mr-[25px] !w-[175px]"
            data-mui-internal-clone-element="true"
          >
            <p
              class="btnActionText"
            >
              NEWWAY20
            </p>
          </button>
        </div>
        <div
          class="resultPicture"
        >
          <img
            alt="helmet"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
